pkg/controller/elasticsearch/user/user_provided.go

```diff
@@ -131,7 +132,13 @@ func retrieveUserProvidedFileRealm(c k8s.Client, es esv1.Elasticsearch, recorder
 			}
 			return filerealm.Realm{}, err
 		}
-		realm, err := filerealm.FromSecret(secret)
+		var realm filerealm.Realm
+		var err error
+		if k8s.HasSecretEntries(secret, corev1.BasicAuthPasswordKey, corev1.BasicAuthUsernameKey) {
```

Comment:
I wonder if using only the keys to decide how the `Secret` should be processed may not lead to situations which could be hard to understand. For example if there's a typo in a key: `kubectl create secret generic my-elastic2-user --from-literal=username=elastic --from-literal=pasword=alongpassword --from-literal=users_roles=role1:my-elastic2-user`, it does not generate any error.

On another hand I don't have any good alternative 🤷 (maybe generate an event if there is an unexpected key ?)

Reply:
I can look into that.